### ISSUE TITLE
InsertMerge/InsertReplace fails if record doesn't exist already in aztables

### ISSUE THREAD

**Issue by kierxn** [Azure/azure-sdk-for-go#16086]
### Bug Report

- Package in question: `github.com/Azure/azure-sdk-for-go/sdk/data/aztables`

- SDK version e.g. `master`, `latest`, `18.1.0`: `github.com/Azure/azure-sdk-for-go/sdk/data/aztables v0.3.0`
- output of `go version`: `go version go1.17.2 windows/amd64`

- What happened?
Trying to batch insert records with `aztables.InsertMerge`/`aztables.InsertReplace` will always return a `Code: ResourceNotFound, Message: The specified resource does not exist.` error.

- What did you expect or want to happen?
I would expect this to insert my record or replace/merge if it exists already.

- How can we reproduce it?
Worth noting that this fails with both InsertMerge and InsertReplace.
```Go
func main() {
	service, _ := aztables.NewServiceClientFromConnectionString(os.Getenv("CONN_STR"), nil)
	client := service.NewClient(os.Getenv("TABLE_NAME"))

	batch := make([]aztables.TransactionAction, 0)

	entity1 := aztables.EDMEntity{
		Entity: aztables.Entity{
			PartitionKey: "p1",
			RowKey:       "r1",
		},
	}
	marshalled, _ := json.Marshal(entity1)

	batch = append(batch, aztables.TransactionAction{
		ActionType: aztables.InsertMerge,
		Entity: marshalled,
	})

	resp, err := client.SubmitTransaction(context.Background(), batch, nil)

	if err != nil {
		panic(err)
	}
	fmt.Println(resp)
}
```

- Anything we should know about your environment.
N/A

**Comment by seankane-msft** [comment]
Hi @kierxn apologies on the bug and thank you for reporting it. I fixed the bug in #16114 just now, barring approval from the team this should go out tomorrow and fix your issue. If you'd like to review the PR I added a test case for this scenario to confirm that it is the same as your issue

**Comment by seankane-msft** [comment]
This has been fixed in `v0.4.0` which is now available in a beta release [here](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/data/aztables).
